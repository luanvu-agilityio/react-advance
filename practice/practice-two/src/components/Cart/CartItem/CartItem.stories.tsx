import type { Meta, StoryObj } from '@storybook/react'
import type { CartItem as CartItemType } from 'types/cart-items'
import CartItem from './CartItem'

const meta: Meta<typeof CartItem> = {
  title: 'Components/Cart/CartItem',
  component: CartItem,
  parameters: {
    layout: 'padded',
    docs: {
      description: {
        component:
          'A cart item component that displays product information, actions, and quantity controls in the shopping cart.',
      },
    },
  },
  argTypes: {
    item: {
      description: 'Cart item data object',
      control: 'object',
    },
    onQuantityChange: {
      description: 'Callback function when quantity changes',
      action: 'quantity changed',
    },
    onUnitChange: {
      description: 'Callback function when unit changes',
      action: 'unit changed',
    },
    onRemove: {
      description: 'Callback function when item is removed',
      action: 'item removed',
    },
  },
  decorators: [
    (Story) => (
      <div
        style={{
          maxWidth: '600px',
          padding: '16px',
          backgroundColor: '#f9f9f9',
        }}
      >
        <Story />
      </div>
    ),
  ],
}

export default meta
type Story = StoryObj<typeof CartItem>

// Sample cart item data
const baseCartItem: CartItemType = {
  id: 1,
  title: 'Fresh Organic Tomatoes',
  description: 'Premium quality organic tomatoes',
  price: 4.99,
  originalPrice: 6.99,
  category: 'Vegetables',
  subcategory: 'Fresh Produce',
  rating: 4,
  imageUrl:
    'https://res.cloudinary.com/ds82onf5q/image/upload/v1748372403/tomato_sample.jpg',
  discountPercentage: 29,
  farm: 'Green Valley Farm',
  freshness: 'New (1 day)',
  delivery: {
    time: '1-2 days',
    location: 'All Countries',
  },
  stock: '50',
  freeShipping: true,
  quantity: 2,
  buyUnit: 'kg',
}

export const Default: Story = {
  args: {
    item: baseCartItem,
    onQuantityChange: (id, quantity) =>
      console.log('Quantity changed:', id, quantity),
    onUnitChange: (id, unit) => console.log('Unit changed:', id, unit),
    onRemove: (id) => console.log('Remove item:', id),
  },
}

export const WithDiscount: Story = {
  args: {
    item: {
      ...baseCartItem,
      title: 'Premium Organic Apples',
      price: 3.99,
      originalPrice: 5.99,
      farm: 'Mountain View Orchard',
      freshness: 'Fresh (2 days)',
      rating: 5,
      discountPercentage: 33,
    },
    onQuantityChange: (id, quantity) =>
      console.log('Quantity changed:', id, quantity),
    onUnitChange: (id, unit) => console.log('Unit changed:', id, unit),
    onRemove: (id) => console.log('Remove item:', id),
  },
  parameters: {
    docs: {
      description: {
        story:
          'Cart item with a discount showing both original and current price',
      },
    },
  },
}

export const NoDiscount: Story = {
  args: {
    item: {
      ...baseCartItem,
      title: 'Fresh Bananas',
      price: 2.49,
      originalPrice: undefined,
      farm: 'Tropical Farms',
      freshness: 'Fresh (1 day)',
      rating: 4,
      discountPercentage: 0,
    },
    onQuantityChange: (id, quantity) =>
      console.log('Quantity changed:', id, quantity),
    onUnitChange: (id, unit) => console.log('Unit changed:', id, unit),
    onRemove: (id) => console.log('Remove item:', id),
  },
  parameters: {
    docs: {
      description: {
        story: 'Cart item without any discount or original price',
      },
    },
  },
}

export const HighQuantity: Story = {
  args: {
    item: {
      ...baseCartItem,
      title: 'Bulk Rice Package',
      quantity: 10,
      buyUnit: 'kg',
      price: 12.99,
      farm: 'Golden Grain Farm',
    },
    onQuantityChange: (id, quantity) =>
      console.log('Quantity changed:', id, quantity),
    onUnitChange: (id, unit) => console.log('Unit changed:', id, unit),
    onRemove: (id) => console.log('Remove item:', id),
  },
  parameters: {
    docs: {
      description: {
        story: 'Cart item with high quantity',
      },
    },
  },
}

export const DifferentUnits: Story = {
  render: () => (
    <div style={{ display: 'flex', flexDirection: 'column', gap: '16px' }}>
      <CartItem
        item={{ ...baseCartItem, buyUnit: 'pcs', title: 'Individual Apples' }}
        onQuantityChange={(id, quantity) =>
          console.log('Quantity changed:', id, quantity)
        }
        onUnitChange={(id, unit) => console.log('Unit changed:', id, unit)}
        onRemove={(id) => console.log('Remove item:', id)}
      />
      <CartItem
        item={{ ...baseCartItem, id: 2, buyUnit: 'kg', title: 'Bulk Potatoes' }}
        onQuantityChange={(id, quantity) =>
          console.log('Quantity changed:', id, quantity)
        }
        onUnitChange={(id, unit) => console.log('Unit changed:', id, unit)}
        onRemove={(id) => console.log('Remove item:', id)}
      />
      <CartItem
        item={{ ...baseCartItem, id: 3, buyUnit: 'lb', title: 'Premium Beef' }}
        onQuantityChange={(id, quantity) =>
          console.log('Quantity changed:', id, quantity)
        }
        onUnitChange={(id, unit) => console.log('Unit changed:', id, unit)}
        onRemove={(id) => console.log('Remove item:', id)}
      />
    </div>
  ),
  parameters: {
    docs: {
      description: {
        story: 'Multiple cart items with different buying units',
      },
    },
  },
}

export const HighRating: Story = {
  args: {
    item: {
      ...baseCartItem,
      title: 'Premium Organic Avocados',
      rating: 5,
      price: 8.99,
      farm: 'California Premium Farms',
      freshness: 'Perfect (today)',
    },
    onQuantityChange: (id, quantity) =>
      console.log('Quantity changed:', id, quantity),
    onUnitChange: (id, unit) => console.log('Unit changed:', id, unit),
    onRemove: (id) => console.log('Remove item:', id),
  },
  parameters: {
    docs: {
      description: {
        story: 'Cart item with perfect 5-star rating',
      },
    },
  },
}

export const InteractiveDemo: Story = {
  args: {
    item: baseCartItem,
    onQuantityChange: (id, quantity) => {
      console.log('Quantity changed:', id, quantity)
      // In a real app, this would update the cart state
    },
    onUnitChange: (id, unit) => {
      console.log('Unit changed:', id, unit)
      // In a real app, this would update the cart state
    },
    onRemove: (id) => {
      console.log('Remove item:', id)
      alert('Item would be removed from cart')
    },
  },
  parameters: {
    docs: {
      description: {
        story:
          'Interactive cart item with working buttons (check console/alerts)',
      },
    },
  },
}
