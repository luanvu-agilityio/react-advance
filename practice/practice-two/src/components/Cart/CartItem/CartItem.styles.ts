import styled from 'styled-components'

export const CartItemContainer = styled.div`
  display: flex;
  gap: 16px;
`

export const ItemImage = styled.div`
  width: 100px;
  height: 67px;
  flex-shrink: 0;

  img {
    width: 100%;
    height: 100%;
    object-fit: cover;
    border-radius: 12px;
  }
`

export const ItemDetails = styled.div`
  flex: 1;
  display: flex;
  flex-direction: column;
  gap: 8px;
`

export const ItemTitle = styled.h3`
  font-size: 15px;
  font-weight: var(--font-weight-medium);
`

export const MetaRow = styled.div`
  display: flex;
  align-items: center;
`

export const MetaLabel = styled.span`
  color: var(--black-shade-3);
  font-size: 12px;
  font-weight: var(--font-weight-regular);
  width: 70px;
  font-family: var(--font-family-secondary);
`

export const MetaValue = styled.span`
  color: var(--black-color-default);
  font-size: 12px;
  font-weight: var(--font-weight-medium);
  font-family: var(--font-family-secondary);
`

export const PriceDisplay = styled.div`
  font-weight: var(--font-weight-semibold);
  color: var(--green-color-default);
  font-size: 18px;
  display: flex;
  flex-direction: column;

  .original-price {
    text-decoration: line-through;
    color: var(--black-shade-2);
    font-size: 12px;
  }
`

export const ItemActions = styled.div`
  display: flex;
  justify-content: space-between;
  align-items: center;
`

export const ActionButtons = styled.div`
  display: flex;
  flex-direction: column;
  gap: 8px;
`

export const ActionButton = styled.button`
  background: none;
  border: none;
  font-family: var(--font-family-secondary);
  font-size: 12px;
  font-weight: var(--font-weight-regular);
  color: var(--black-shade-2);
  cursor: pointer;
  display: flex;
  align-items: center;
  gap: 6px;
  padding: 0;

  img {
    width: 12px;
    height: 12px;
  }

  &:hover {
    color: var(--black-color-default);
  }
`
