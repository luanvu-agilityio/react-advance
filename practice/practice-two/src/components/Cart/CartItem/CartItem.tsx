'use client'

import { renderStars } from '@helpers/renderStar'
import BuyingUnit from '@components/common/BuyingUnit/BuyingUnit'
import type { CartItem as CartItemType } from 'types/cart-items'
import {
  ActionButton,
  ActionButtons,
  CartItemContainer,
  ItemActions,
  ItemDetails,
  ItemImage,
  ItemTitle,
  MetaLabel,
  MetaRow,
  MetaValue,
  PriceDisplay,
} from './CartItem.styles'

interface CartItemProps {
  item: CartItemType
  onQuantityChange: (id: number, quantity: number) => void
  onUnitChange: (id: number, buyUnit: string) => void
  onRemove: (id: number) => void
}

const CartItem = ({
  item,
  onQuantityChange,
  onUnitChange,
  onRemove,
}: CartItemProps) => {
  return (
    <CartItemContainer>
      <div
        style={{
          display: 'flex',
          flexDirection: 'column',
          gap: '12px',
        }}
      >
        <ItemImage>
          <img src={item.imageUrl} alt={item.title} />
        </ItemImage>
        <ActionButtons>
          <ActionButton>
            <img
              src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372436/wishlist_cltlcf.svg"
              alt="Wishlist"
            />
            Wishlist
          </ActionButton>
          <ActionButton>
            <img
              src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372430/compare_oeiv1m.svg"
              alt="Compare"
            />
            Compare
          </ActionButton>
          <ActionButton onClick={() => onRemove(item.id)}>
            <img
              src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372432/remove_gbwco0.svg"
              alt="Remove"
            />
            Remove
          </ActionButton>
        </ActionButtons>
      </div>

      <ItemDetails>
        <div
          style={{
            display: 'flex',
            flexDirection: 'column',
            gap: '4px',
          }}
        >
          <ItemTitle>{item.title}</ItemTitle>
          <div
            style={{
              display: 'flex',
              flexDirection: 'column',
              gap: '4px',
            }}
          >
            <MetaRow>
              <MetaLabel>Farm:</MetaLabel>
              <MetaValue>{item.farm}</MetaValue>
            </MetaRow>

            <MetaRow>
              <MetaLabel>Freshness:</MetaLabel>
              <MetaValue>{item.freshness}</MetaValue>
            </MetaRow>
          </div>
          {renderStars(item.rating ?? 0)}
        </div>
        <ItemActions>
          <PriceDisplay>
            {item.price.toFixed(2)} USD
            {item.originalPrice && item.originalPrice > item.price && (
              <span className="original-price">
                {item.originalPrice.toFixed(2)} USD
              </span>
            )}
          </PriceDisplay>

          <BuyingUnit
            quantity={item.quantity}
            unit={item.buyUnit}
            onQuantityChange={(value) => onQuantityChange(item.id, value)}
            onUnitChange={(value) => onUnitChange(item.id, value)}
            size="small"
          />
        </ItemActions>
      </ItemDetails>
    </CartItemContainer>
  )
}

export default CartItem
