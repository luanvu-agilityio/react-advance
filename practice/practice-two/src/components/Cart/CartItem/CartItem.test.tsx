import { render, screen, fireEvent } from '@testing-library/react'

import type { CartItem as CartItemType } from 'types/cart-items'
import CartItem from './CartItem'

jest.mock('@helpers/renderStar', () => ({
  renderStars: (rating: number) => (
    <div data-testid="star-rating">{rating} stars</div>
  ),
}))

jest.mock('@components/common/BuyingUnit/BuyingUnit', () => {
  return function MockBuyingUnit({
    quantity,
    unit,
    onQuantityChange,
    onUnitChange,
    size,
  }: {
    quantity: number
    unit: string
    onQuantityChange: (quantity: number) => void
    onUnitChange: (unit: string) => void
    size?: 'small' | 'medium' | 'large'
  }) {
    return (
      <div data-testid="buying-unit" data-size={size}>
        <button
          data-testid="decrease-quantity"
          onClick={() => onQuantityChange(quantity - 1)}
        >
          -
        </button>
        <span data-testid="quantity-display">
          {quantity} {unit}
        </span>
        <button
          data-testid="increase-quantity"
          onClick={() => onQuantityChange(quantity + 1)}
        >
          +
        </button>
        <select
          data-testid="unit-selector"
          value={unit}
          onChange={(e) => onUnitChange(e.target.value)}
        >
          <option value="pcs">pcs</option>
          <option value="kg">kg</option>
          <option value="lb">lb</option>
        </select>
      </div>
    )
  }
})

describe('CartItem Component', () => {
  const mockCartItem: CartItemType = {
    id: 1,
    title: 'Test Product',
    description: 'Test description',
    price: 9.99,
    originalPrice: 12.99,
    category: 'Test Category',
    subcategory: 'Test Subcategory',
    rating: 4,
    discountPercentage: 23,
    farm: 'Test Farm',
    freshness: 'Fresh (1 day)',
    delivery: {
      time: '1-2 days',
      location: 'All Countries',
    },
    stock: '50',
    freeShipping: true,
    quantity: 2,
    buyUnit: 'kg',
    images: { main: 'test-image.jpg', gallery: ['test-image.jpg'] }, // Add a mock image object
  }

  const mockProps = {
    item: mockCartItem,
    onQuantityChange: jest.fn(),
    onUnitChange: jest.fn(),
    onRemove: jest.fn(),
  }

  beforeEach(() => {
    jest.clearAllMocks()
  })

  it('matches the snapshot for basic rendering', () => {
    const { container } = render(<CartItem {...mockProps} />)
    expect(container).toMatchSnapshot()
  })

  // Add snapshots for different states
  it('matches snapshot without original price', () => {
    const itemWithoutOriginalPrice = {
      ...mockCartItem,
      originalPrice: undefined,
    }
    const { container } = render(
      <CartItem {...mockProps} item={itemWithoutOriginalPrice} />
    )
    expect(container).toMatchSnapshot()
  })

  it('matches snapshot with different unit', () => {
    const itemWithPcsUnit = {
      ...mockCartItem,
      buyUnit: 'pcs',
    }
    const { container } = render(
      <CartItem {...mockProps} item={itemWithPcsUnit} />
    )
    expect(container).toMatchSnapshot()
  })

  it('matches snapshot with high quantity', () => {
    const itemWithHighQuantity = {
      ...mockCartItem,
      quantity: 999,
    }
    const { container } = render(
      <CartItem {...mockProps} item={itemWithHighQuantity} />
    )
    expect(container).toMatchSnapshot()
  })

  it('calls onRemove when remove button is clicked', () => {
    render(<CartItem {...mockProps} />)
    const removeButton = screen.getByText('Remove')
    fireEvent.click(removeButton)
    expect(mockProps.onRemove).toHaveBeenCalledWith(1)
  })

  it('calls onQuantityChange when quantity is modified', () => {
    render(<CartItem {...mockProps} />)
    const increaseButton = screen.getByTestId('increase-quantity')
    fireEvent.click(increaseButton)
    expect(mockProps.onQuantityChange).toHaveBeenCalledWith(1, 3)

    const decreaseButton = screen.getByTestId('decrease-quantity')
    fireEvent.click(decreaseButton)
    expect(mockProps.onQuantityChange).toHaveBeenCalledWith(1, 1)
  })

  it('calls onUnitChange when unit is modified', () => {
    render(<CartItem {...mockProps} />)
    const unitSelector = screen.getByTestId('unit-selector')
    fireEvent.change(unitSelector, { target: { value: 'pcs' } })
    expect(mockProps.onUnitChange).toHaveBeenCalledWith(1, 'pcs')
  })
})
