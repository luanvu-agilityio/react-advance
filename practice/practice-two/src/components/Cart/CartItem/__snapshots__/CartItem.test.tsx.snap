// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CartItem Component matches snapshot with different unit 1`] = `
<div>
  <div
    class="sc-Qotzb kCEAoD"
  >
    <div
      style="display: flex; flex-direction: column; gap: 12px;"
    >
      <div
        class="sc-fYsHOw eooCtm"
      >
        <img
          alt="Test Product"
          src="https://test-image.jpg"
        />
      </div>
      <div
        class="sc-jMsorb loeJqH"
      >
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Wishlist"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372436/wishlist_cltlcf.svg"
          />
          Wishlist
        </button>
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Compare"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372430/compare_oeiv1m.svg"
          />
          Compare
        </button>
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Remove"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372432/remove_gbwco0.svg"
          />
          Remove
        </button>
      </div>
    </div>
    <div
      class="sc-dKREkF hyvUmY"
    >
      <div
        style="display: flex; flex-direction: column; gap: 4px;"
      >
        <h3
          class="sc-fWnslK hLsXWl"
        >
          Test Product
        </h3>
        <div
          style="display: flex; flex-direction: column; gap: 4px;"
        >
          <div
            class="sc-dIMoHT gqZOSq"
          >
            <span
              class="sc-iQQCXo hELsaQ"
            >
              Farm:
            </span>
            <span
              class="sc-gDpztx ioyOiP"
            >
              Test Farm
            </span>
          </div>
          <div
            class="sc-dIMoHT gqZOSq"
          >
            <span
              class="sc-iQQCXo hELsaQ"
            >
              Freshness:
            </span>
            <span
              class="sc-gDpztx ioyOiP"
            >
              Fresh (1 day)
            </span>
          </div>
        </div>
        <div
          data-testid="star-rating"
        >
          4
           stars
        </div>
      </div>
      <div
        class="sc-icnseD dRzQnJ"
      >
        <div
          class="sc-kpOvIu brXGTR"
        >
          9.99
           USD
          <span
            class="original-price"
          >
            12.99
             USD
          </span>
        </div>
        <div
          data-size="small"
          data-testid="buying-unit"
        >
          <button
            data-testid="decrease-quantity"
          >
            -
          </button>
          <span
            data-testid="quantity-display"
          >
            2
             
            pcs
          </span>
          <button
            data-testid="increase-quantity"
          >
            +
          </button>
          <select
            data-testid="unit-selector"
          >
            <option
              value="pcs"
            >
              pcs
            </option>
            <option
              value="kg"
            >
              kg
            </option>
            <option
              value="lb"
            >
              lb
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CartItem Component matches snapshot with high quantity 1`] = `
<div>
  <div
    class="sc-Qotzb kCEAoD"
  >
    <div
      style="display: flex; flex-direction: column; gap: 12px;"
    >
      <div
        class="sc-fYsHOw eooCtm"
      >
        <img
          alt="Test Product"
          src="https://test-image.jpg"
        />
      </div>
      <div
        class="sc-jMsorb loeJqH"
      >
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Wishlist"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372436/wishlist_cltlcf.svg"
          />
          Wishlist
        </button>
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Compare"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372430/compare_oeiv1m.svg"
          />
          Compare
        </button>
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Remove"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372432/remove_gbwco0.svg"
          />
          Remove
        </button>
      </div>
    </div>
    <div
      class="sc-dKREkF hyvUmY"
    >
      <div
        style="display: flex; flex-direction: column; gap: 4px;"
      >
        <h3
          class="sc-fWnslK hLsXWl"
        >
          Test Product
        </h3>
        <div
          style="display: flex; flex-direction: column; gap: 4px;"
        >
          <div
            class="sc-dIMoHT gqZOSq"
          >
            <span
              class="sc-iQQCXo hELsaQ"
            >
              Farm:
            </span>
            <span
              class="sc-gDpztx ioyOiP"
            >
              Test Farm
            </span>
          </div>
          <div
            class="sc-dIMoHT gqZOSq"
          >
            <span
              class="sc-iQQCXo hELsaQ"
            >
              Freshness:
            </span>
            <span
              class="sc-gDpztx ioyOiP"
            >
              Fresh (1 day)
            </span>
          </div>
        </div>
        <div
          data-testid="star-rating"
        >
          4
           stars
        </div>
      </div>
      <div
        class="sc-icnseD dRzQnJ"
      >
        <div
          class="sc-kpOvIu brXGTR"
        >
          9.99
           USD
          <span
            class="original-price"
          >
            12.99
             USD
          </span>
        </div>
        <div
          data-size="small"
          data-testid="buying-unit"
        >
          <button
            data-testid="decrease-quantity"
          >
            -
          </button>
          <span
            data-testid="quantity-display"
          >
            999
             
            kg
          </span>
          <button
            data-testid="increase-quantity"
          >
            +
          </button>
          <select
            data-testid="unit-selector"
          >
            <option
              value="pcs"
            >
              pcs
            </option>
            <option
              value="kg"
            >
              kg
            </option>
            <option
              value="lb"
            >
              lb
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CartItem Component matches snapshot without original price 1`] = `
<div>
  <div
    class="sc-Qotzb kCEAoD"
  >
    <div
      style="display: flex; flex-direction: column; gap: 12px;"
    >
      <div
        class="sc-fYsHOw eooCtm"
      >
        <img
          alt="Test Product"
          src="https://test-image.jpg"
        />
      </div>
      <div
        class="sc-jMsorb loeJqH"
      >
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Wishlist"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372436/wishlist_cltlcf.svg"
          />
          Wishlist
        </button>
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Compare"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372430/compare_oeiv1m.svg"
          />
          Compare
        </button>
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Remove"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372432/remove_gbwco0.svg"
          />
          Remove
        </button>
      </div>
    </div>
    <div
      class="sc-dKREkF hyvUmY"
    >
      <div
        style="display: flex; flex-direction: column; gap: 4px;"
      >
        <h3
          class="sc-fWnslK hLsXWl"
        >
          Test Product
        </h3>
        <div
          style="display: flex; flex-direction: column; gap: 4px;"
        >
          <div
            class="sc-dIMoHT gqZOSq"
          >
            <span
              class="sc-iQQCXo hELsaQ"
            >
              Farm:
            </span>
            <span
              class="sc-gDpztx ioyOiP"
            >
              Test Farm
            </span>
          </div>
          <div
            class="sc-dIMoHT gqZOSq"
          >
            <span
              class="sc-iQQCXo hELsaQ"
            >
              Freshness:
            </span>
            <span
              class="sc-gDpztx ioyOiP"
            >
              Fresh (1 day)
            </span>
          </div>
        </div>
        <div
          data-testid="star-rating"
        >
          4
           stars
        </div>
      </div>
      <div
        class="sc-icnseD dRzQnJ"
      >
        <div
          class="sc-kpOvIu brXGTR"
        >
          9.99
           USD
        </div>
        <div
          data-size="small"
          data-testid="buying-unit"
        >
          <button
            data-testid="decrease-quantity"
          >
            -
          </button>
          <span
            data-testid="quantity-display"
          >
            2
             
            kg
          </span>
          <button
            data-testid="increase-quantity"
          >
            +
          </button>
          <select
            data-testid="unit-selector"
          >
            <option
              value="pcs"
            >
              pcs
            </option>
            <option
              value="kg"
            >
              kg
            </option>
            <option
              value="lb"
            >
              lb
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CartItem Component matches the snapshot for basic rendering 1`] = `
<div>
  <div
    class="sc-Qotzb kCEAoD"
  >
    <div
      style="display: flex; flex-direction: column; gap: 12px;"
    >
      <div
        class="sc-fYsHOw eooCtm"
      >
        <img
          alt="Test Product"
          src="https://test-image.jpg"
        />
      </div>
      <div
        class="sc-jMsorb loeJqH"
      >
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Wishlist"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372436/wishlist_cltlcf.svg"
          />
          Wishlist
        </button>
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Compare"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372430/compare_oeiv1m.svg"
          />
          Compare
        </button>
        <button
          class="sc-cXawGu gGzVVD"
        >
          <img
            alt="Remove"
            src="https://res.cloudinary.com/ds82onf5q/image/upload/v1748372432/remove_gbwco0.svg"
          />
          Remove
        </button>
      </div>
    </div>
    <div
      class="sc-dKREkF hyvUmY"
    >
      <div
        style="display: flex; flex-direction: column; gap: 4px;"
      >
        <h3
          class="sc-fWnslK hLsXWl"
        >
          Test Product
        </h3>
        <div
          style="display: flex; flex-direction: column; gap: 4px;"
        >
          <div
            class="sc-dIMoHT gqZOSq"
          >
            <span
              class="sc-iQQCXo hELsaQ"
            >
              Farm:
            </span>
            <span
              class="sc-gDpztx ioyOiP"
            >
              Test Farm
            </span>
          </div>
          <div
            class="sc-dIMoHT gqZOSq"
          >
            <span
              class="sc-iQQCXo hELsaQ"
            >
              Freshness:
            </span>
            <span
              class="sc-gDpztx ioyOiP"
            >
              Fresh (1 day)
            </span>
          </div>
        </div>
        <div
          data-testid="star-rating"
        >
          4
           stars
        </div>
      </div>
      <div
        class="sc-icnseD dRzQnJ"
      >
        <div
          class="sc-kpOvIu brXGTR"
        >
          9.99
           USD
          <span
            class="original-price"
          >
            12.99
             USD
          </span>
        </div>
        <div
          data-size="small"
          data-testid="buying-unit"
        >
          <button
            data-testid="decrease-quantity"
          >
            -
          </button>
          <span
            data-testid="quantity-display"
          >
            2
             
            kg
          </span>
          <button
            data-testid="increase-quantity"
          >
            +
          </button>
          <select
            data-testid="unit-selector"
          >
            <option
              value="pcs"
            >
              pcs
            </option>
            <option
              value="kg"
            >
              kg
            </option>
            <option
              value="lb"
            >
              lb
            </option>
          </select>
        </div>
      </div>
    </div>
  </div>
</div>
`;
